en/peripherals/gripper.md

```diff
@@ -0,0 +1,107 @@
+
+:::note
+The [Roboclaw driver](../modules/modules_driver.md#roboclaw) isn't supported as a Gripper yet.
+However in the future it may get supported as a Gripper.
+:::
+
+## Using a Gripper
+
+For information on using a gripper in missions see [Payload Delivery Missions](../flying/package_delivery_mission.md).
+
+You can trigger the gripper manually if you've mapped `gripper open` and `gripper close` buttons in the [QGC Joystick Configuration](#qgc-joystick-configuration).
+Note that if you press **Grab** button while the gripper is opening, it will automatically abort releasing behavior and go to the Close position, effectively cancelling the release command.
+If you do this in a mission while the release is actually happening then the [delivery will be cancelled](../flying/package_delivery_mission.md#manual-control-of-gripper-in-missions).
+
+MAVLink applications, such as ground stations, can also control the gripper using the [MAV_CMD_DO_GRIPPER](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_GRIPPER) MAVLink command.
+
+
+## PX4 Configuration
+
+### Package Delivery Configuration
+
+PX4 Gripper support is tied to the package delivery feature, which must be enabled and configured in order to be able to use a gripper.
+
+1. Set [`PD_GRIPPER_EN`](../advanced_config/parameter_reference.md#PD_GRIPPER_EN) parameter to 1 (reboot required after change).
+1. Set [`PD_GRIPPER_TYPE`](../advanced_config/parameter_reference.md#PD_GRIPPER_TYPE) to match your Gripper.
+   For example, set to `Servo` for a [Servo Gripper](gripper_servo.md).
+
+### Gripper Actuator Mapping
+
+To enable the output of the Gripper, set the Function of the output port where the gripper is connected to as `Gripper` in the [Actuators](../config/actuators.md#actuator-outputs) tab.
+
+<!-- Replace this with actuator diagram -->
+
+### QGC Joystick Configuration
+
+QGroundControl [Joystick](../config/joystick.md) configuration allows you to map gripper actions to Joystick buttons, after which you will be open and close the gripper manually.
+
+Open the Joystick configuration by selecting: **QGC Logo (upper-left) > Vehicle Setup > Joystick > Button Assignment** tab.
+Select `Gripper Open` and `Gripper Close` actions for your desired joystick buttons
+
+![Gripper action mapping](../../assets/config/gripper/qgc_gripper_actions_joystick.png)
+
+You can test the actions by clicking on the mapped buttons and checking for gripper movement.
+If the gripper doesn't move as expected check the package delivery configuration and actuator mapping are set up properly.
+
+### Enable Pre-ARM Mode
+
+We need to set a parameter to enable [pre-arming](../advanced_config/prearm_arm_disarm.md).
+This keeps the motors disabled but allows the gripper to be opened and closed for attaching the payload (avoiding potential danger from spinning propellers).
+
+Set [`COM_PREARM_MODE`](../advanced_config/parameter_reference.md#COM_PREARM_MODE) to `Always`.
```

Comment:
Do we need a caveat or warning, and if so what? I.e. if it is completely safe to always set prearm mode, then why isn't this always enabled or would default-on. My thinking is that there are case where prearm might be dangerous.

Reply:
I have asked @MaEtUgR about it!

Reply:
Thanks @junwoo091400 - What did he say?

Reply:
I said I so far always enabled prearm mode especially for preflight surface checks. I'm guessing it was previously disabled by default such that if you accidentally plug a motor on a servo pin the chances are lower that it spins up. But that assumption is flawed since servos usually have 1500 as disarmed values. Also assignment of actuators is now much better with control allocation. So the only remaining thing I see is to save the servos from extra work while the vehicle is powered on the bench 🤷‍♂️ But a fixed-wing pro like @sfuhrer might have more insights.

Reply:
@MaEtUgR Thanks. IMO we don't particularly need a warning. As you say, prearm is only unsafe if somehow motors or "dangerous actuators" can move in this state. So possibly risky on first setup, but not for use once properly configured.

Reply:
Yes, exactly 👍

Reply:
Context: https://github.com/PX4/PX4-Autopilot/pull/17309#issuecomment-992205444
TL;DR: I think we shouldn't just change the default to "Always", but stay on the safe side.
A user then should make the deliberate decision (after setting up the vehicle the first time and testing the wiring) whether one wants to have the servos already enabled by without arming or in prearm mode. But in general "arming" should be active for all actuators, not just the motors.

Reply:
Fair enough. I think at this point you've probably already verified your motors and servos are set up properly and prearm is safe. So no need for a warning. No harm on being on the safe side for the setting either.